blockchain/blockchain.go

```diff
@@ -627,10 +636,31 @@ func (bc *BlockChain) setHeadBeyondRoot(head uint64, root common.Hash, repair bo
 		}
 	}
 
+	// Delete istanbul snapshot database further two epochs
+	var (
+		curBlkNum       = bc.CurrentBlock().Number().Uint64()
+		epoch           = bc.Config().Istanbul.Epoch
+		twoEpochs       = epoch * 2
+		beforeTwoEpochs uint64
+	)
+	if twoEpochs > curBlkNum {
+		beforeTwoEpochs = 1
+	} else {
+		beforeTwoEpochs = curBlkNum - twoEpochs
+	}
+	for i := curBlkNum; i >= beforeTwoEpochs; i-- {
```

Comment:
I'm wondering why the for loop's condition is `i >= beforeTwoEpochs`.

Assume there's a vote block `V` (that contains vote data) and a governance block `G` (that contains governance data).
For `setHeadBeyondRoot(x)`,
- if `x < V` or `x >= G`: no problem
- if `V <= x < G`: bad block at `G`

I think this logic addresses the second condition.
Then, shouldn't the condition of this for loop be `i >= V`?
If it's hard to fetch `V`, I'm thinking `x - ( x % epoch)` would do (which will be less than `V`).

Reply:
A smarter comment. Thank you. Updated. Thank you for clearly pinpointing the problem that PR addresses.